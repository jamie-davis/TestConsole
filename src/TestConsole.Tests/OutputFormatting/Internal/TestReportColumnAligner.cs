using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using TestConsole.OutputFormatting.Internal;
using TestConsole.Tests.TestingUtilities;

namespace TestConsole.Tests.OutputFormatting.Internal
{
    [TestFixture]
    [UseReporter(typeof (CustomReporter))]
    public class TestReportColumnAligner
    {
        [Test]
        public void SingleLineIsFormatted()
        {
            var output = ReportColumnAligner.AlignColumns(new[] {10, 3, 5},
                new[] {new[] {"ten widexx"}, new[] {"3wd"}, new[] {"fivew"}});
            Console.WriteLine(output);
            Approvals.Verify(output);
        }

        [Test]
        public void ExcessiveDataIsTrimmed()
        {
            var output = ReportColumnAligner.AlignColumns(new[] { 8, 2, 4 }, 
                new[] { new[] { "ten widexx" }, new[] { "3wd" }, new[] { "fivew" } });
            Console.WriteLine(output);
            Approvals.Verify(output);
        }

        [Test]
        public void ShortDataIsPadded()
        {
            var output = ReportColumnAligner.AlignColumns(new[] { 14, 6, 7 }, 
                new[] { new[] { "ten widexx" }, new[] { "3wd" }, new[] { "fivew" } });
            Console.WriteLine(output);
            Approvals.Verify(output);
        }

        [Test]
        public void MultiRowDataIsFormattedInColumns()
        {
            var output = ReportColumnAligner.AlignColumns(new[] { 14, 6, 7 }, 
                new[] { new[] { "ten widexx", "more", "too wide too wide" }, new[] { "3wd", "3mo", "1" }, new[] { "fivew", "again", "again" } });
            Console.WriteLine(output);
            Approvals.Verify(output);
        }

        [Test]
        public void InconsistentRowsAreTopAligned()
        {
            var output = ReportColumnAligner.AlignColumns(new[] { 10, 3, 5 }, 
                new[] { new[] { "ten widexx", "more" }, new[] { "3wd"}, new[] { "fivew", "again", "again", "again" } });
            Console.WriteLine(output);
            Approvals.Verify(output);
        }

        [Test]
        public void InconsistentRowsAreBottomAligned()
        {
            var output = ReportColumnAligner.AlignColumns(new[] { 10, 3, 5 }, 
                new[] { new[] { "ten widexx", "more" }, new[] { "3wd"}, new[] { "fivew", "again", "again", "again" } },
                ColVerticalAligment.Bottom);
            Console.WriteLine(output);
            Approvals.Verify(output);
        }
    }
}