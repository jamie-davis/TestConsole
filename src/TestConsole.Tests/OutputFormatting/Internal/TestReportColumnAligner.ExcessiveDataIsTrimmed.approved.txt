ten wide 3w five
